sdk/src/main/java/com/mapbox/maps/renderer/MapboxWidgetRenderer.kt

```diff
@@ -0,0 +1,171 @@
+package com.mapbox.maps.renderer
+
+import android.opengl.GLES20
+import com.mapbox.common.Logger
+import com.mapbox.maps.renderer.egl.EGLCore
+import com.mapbox.maps.renderer.widget.Widget
+import java.util.concurrent.CopyOnWriteArraySet
+import javax.microedition.khronos.egl.EGLContext
+import javax.microedition.khronos.egl.EGLSurface
+
+internal class MapboxWidgetRenderer(
+  private val translucentSurface: Boolean,
+  private val antialiasingSampleCount: Int,
+) {
+  private var widgetEglPrepared = false
+  private lateinit var widgetEglSurface: EGLSurface
+  private lateinit var widgetEglCore: EGLCore
+
+  private val textures = intArrayOf(0)
+  private val framebuffers = intArrayOf(0)
+
+  private val widgets = CopyOnWriteArraySet<Widget>()
```

Comment:
why copy on write?

Reply:
Was thinking about the multithreading and adding safety guarantees, but seems unneeded indeed. Removing

Reply:
not see it being removed still